Description: external module: "scriptname" local variable added
Author: vir@ctm.ru
---
"scriptname" local variable added to allow tcp-connected global script to set it's name.
Also, name can be set from %%>connect. In that case script name is set from "id" optional paramter.

Index: yate/modules/extmodule.cpp
===================================================================
--- yate.orig/modules/extmodule.cpp	2016-01-14 16:34:53.000000000 +0300
+++ yate/modules/extmodule.cpp	2016-01-15 15:06:17.000000000 +0300
@@ -1343,6 +1343,10 @@
 		role.c_str(),chan.c_str(),type.c_str());
 	    if (role == "global") {
 		m_role = RoleGlobal;
+		if(! chan.null()) {
+		    Debug(DebugMild,"Renamed external '%s' to '%s'",m_script.safe(),chan.c_str());
+		    m_script = chan;
+		}
 		return false;
 	    }
 	    else if (role == "channel") {
@@ -1471,6 +1475,14 @@
 		    m_chan->setId(val);
 		ok = true;
 	    }
+	    else if (id == "scriptname") {
+		if (val.null())
+		    val = m_script;
+		else {
+		    Debug(DebugMild,"Renamed external '%s' to '%s'",m_script.safe(),val.c_str());
+		    m_script = val;
+		}
+	    }
 	    else if (m_chan && (id == "disconnected")) {
 		m_chan->setDisconn(val.toBoolean(m_chan->disconn()));
 		val = m_chan->disconn();
